src/BlockTypes/MiniCart.php

```diff
@@ -442,4 +442,34 @@ protected function get_tax_label() {
 	protected function get_cart_payload() {
 		return WC()->api->get_endpoint_data( '/wc/store/cart' );
 	}
+
+	/**
+	 * Prepare translations for inner blocks and dependencies.
+	 */
+	protected function get_dependencies_translations() {
+		$wp_scripts   = wp_scripts();
+		$translations = array();
+
+		$blocks = [
+			'mini-cart-contents-block/filled-cart',
+			'mini-cart-contents-block/empty-cart',
+			'mini-cart-contents-block/title',
+			'mini-cart-contents-block/items',
+			'mini-cart-contents-block/products-table',
+			'mini-cart-contents-block/footer',
+			'mini-cart-contents-block/shopping-button',
+		];
+		$chunks = preg_filter( '/$/', '-frontend', $blocks );
+
+		foreach ( $chunks as $chunk ) {
+			$handle = 'wc-blocks-' . $chunk . '-chunk';
+			$this->asset_api->register_script( $handle, $this->asset_api->get_block_asset_build_path( $chunk ), [], true );
+			$translations[] = $wp_scripts->print_translations( $handle, false );
+			wp_deregister_script( $handle );
+		}
+
+		$translations = array_filter( $translations );
+
+		return implode( '', $translations );
+	}
```

Comment:
```suggestion
		/**
		 * Temporary remove the this filter so $wp_scripts->print_translations
		 * calls won't accident print the translations scripts for the block.
		 *
		 * $wp_scripts->print_translations() calls load_script_textdomain()
		 * which calls load_script_translations() containing the below filter.
		 */
		remove_filter( 'pre_load_script_translations', 'woocommerce_blocks_get_i18n_data_json', 10, 4 );

		$wp_scripts   = wp_scripts();
		$translations = array();

		$blocks = [
			'mini-cart-contents-block/filled-cart',
			'mini-cart-contents-block/empty-cart',
			'mini-cart-contents-block/title',
			'mini-cart-contents-block/items',
			'mini-cart-contents-block/products-table',
			'mini-cart-contents-block/footer',
			'mini-cart-contents-block/shopping-button',
		];
		$chunks = preg_filter( '/$/', '-frontend', $blocks );

		foreach ( $chunks as $chunk ) {
			$handle = 'wc-blocks-' . $chunk . '-chunk';
			$this->asset_api->register_script( $handle, $this->asset_api->get_block_asset_build_path( $chunk ), [], true );
			$translations[] = $wp_scripts->print_translations( $handle, false );
			wp_deregister_script( $handle );
		}

		foreach ( array_keys( $this->scripts_to_lazy_load ) as $script ) {
			$translations[] = $wp_scripts->print_translations( $script, false );
		}

		$translations = array_filter( $translations );

		// Re-add the filter.
		add_filter( 'pre_load_script_translations', 'woocommerce_blocks_get_i18n_data_json', 10, 4 );

		return implode( '', $translations );
	}
```

Reply:
This code adds translations of Mini Cart dependencies back without throwing the JS error on the non-block themes. Huge props @Aljullu for the suggestion!

Reply:
Also, we should review `woocommerce_blocks_get_i18n_data_json()`. That function should only return value, but it contains a `printf` call which causes the issue we're solving here.

Reply:
@gigitux I corrected the suggestion in case you checked it.